### ISSUE TITLE
[Discuss] Easy Way to Deploy in Single/Cluster Mode.

### ISSUE THREAD

**Issue by swartz-k** [opensearch-project/OpenSearch#561]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Support an easy way to start OpenSearch in local machine or K8S, such as docker compose、Helm Chart or something

**Describe alternatives you've considered**
Support Helm Chart and docker deploy docs

**Comment by stockholmux** [comment]
@swartz-k This is for Open Distro, but would something like the steps listed [here](https://opendistro.github.io/for-elasticsearch/downloads.html#try) meet your need if they applied to OpenSearch?

**Comment by swartz-k** [comment]
> @swartz-k This is for Open Distro, but would something like the steps listed [here](https://opendistro.github.io/for-elasticsearch/downloads.html#try) meet your need if they applied to OpenSearch?

Sure. But if possible we wanna deploy it on k8s, so Helm may be needed too. And also we would like to contribute and make this project more successful.

**Comment by iron-rain** [comment]
Is anyone actively working on this? I have put together a chart that deploys a cluster mode OpenSearch instance. 
Happy to neaten it up and submit a PR if it's desired?

**Comment by stockholmux** [comment]
@iron-rain I would put together a PR. I'm not sure if it would ultimately end up in this repo or not though.

**Comment by abbashus** [comment]
@iron-rain I think [opensearch-project/opensearch-devops ](https://github.com/opensearch-project/opensearch-devops) would be the right place to raise a PR.

**Comment by LaurentDumont** [comment]
Does anyone know what will happen with the images for Opendistro on the Amazon Docker hub? I'd like to leverage Opendistro but don't really want to deploy it and get the rug pulled from under me ;)

**Comment by iron-rain** [comment]
@stockholmux @abbashus Thanks! Have a raised a PR to get some discussion going hopefully.

**Comment by dblock** [comment]
Does this issue belong in https://github.com/opensearch-project/opensearch-devops?

**Comment by TheAlgo** [comment]
@dblock We already have a light-weight chart for deploying  in a single/Cluster mode which is highly customisable. 
PR: https://github.com/opensearch-project/opensearch-devops/pull/4

**Comment by dblock** [comment]
So should we just close this? @swartz-k does that do everything you need it to do?

**Comment by ekarlso** [comment]
@TheAlgo  It seems that the new chart though is more primitive then the older one that has support for splitting up cluster into different roles. Any idea on why simply the old chart from ODFE was not copied?

**Comment by TheAlgo** [comment]
> @TheAlgo It seems that the new chart though is more primitive then the older one that has support for splitting up cluster into different roles. Any idea on why simply the old chart from ODFE was not copied?

There were 2 major reasons:
1. Provide more complexity for spinning up the clusters.
2. Make the charts monolithic by excluding OpenSearch Dashboards from it.

**Comment by ekarlso** [comment]
@TheAlgo can you still split roles into data, client and master?

**Comment by TheAlgo** [comment]
> @TheAlgo can you still split roles into data, client and master?

Yes we can do it by installing three charts with different values. We can set different role each time you install a chart. There are plans to provide these examples in the repo itself.

**Comment by ekarlso** [comment]
Unrelated q, is there any plans for a Operator for OS?

**Comment by TheAlgo** [comment]
> Unrelated q, is there any plans for a Operator for OS?

I see there is an open [issue](https://github.com/opensearch-project/opensearch-devops/issues/13). Not sure about the plans , @peterzhuamazon will be thr right person to comment on it.

**Comment by ermukesh09** [comment]
Is this ok to run k8s cluster using helm in a production environment?
